lib/transaction.js

```diff
@@ -26,6 +30,19 @@ var Transaction = module.exports = function(sequelize, options) {
   } else {
     this.id = this.name = Utils.generateUUID();
   }
+
+  // `transaction.promise` implementation #2805
+  // main propose is to allow chainable logic inside hooks
+  this.promise = new Promise(function () {});
```

Comment:
In this case, we don't have a *resolver*, instead, we resolve promise using `.commit` or `.rollback` method calling, so i guess we can go around by calling *bluebird* 's internal API.
I'm still with doubts between: calling `promise._resolveFromResolver`, bind directly last returned promise with `finally` or calling callbacks (`._resolveCallback` and `._rejectCallback`) within a `.tap`